common/phal_L4/spi/spi.c

```diff
@@ -95,11 +112,28 @@ bool PHAL_SPI_transfer(SPI_InitConfig_t* spi, const uint8_t* out_data, const uin
     PHAL_startTxfer(spi->rx_dma_cfg);
 
     // Enable the DMA IRQ
-    NVIC_EnableIRQ(DMA1_Channel2_IRQn);
-    NVIC_EnableIRQ(DMA1_Channel3_IRQn);
+    if (spi->periph == SPI1) {
+        NVIC_EnableIRQ(DMA1_Channel2_IRQn);
+        NVIC_EnableIRQ(DMA1_Channel3_IRQn);
+    }
+    #ifdef SPI2
+        else if (spi->periph == SPI2) {
+            NVIC_EnableIRQ(DMA1_Channel4_IRQn);
+            NVIC_EnableIRQ(DMA1_Channel5_IRQn);
+        }
+    #endif
+    else if (spi->periph == SPI3) {
+        NVIC_EnableIRQ(DMA2_Channel1_IRQn);
+        NVIC_EnableIRQ(DMA2_Channel2_IRQn);
+    }
+    //Defaults to SPI1
+    else {
+        NVIC_EnableIRQ(DMA1_Channel2_IRQn);
+        NVIC_EnableIRQ(DMA1_Channel3_IRQn);
+    }
 
     // Start transaction
-    SPI1->CR1 |= SPI_CR1_SPE;
+    spi->periph->CR1 |= SPI_CR1_SPE;
 
     // STM32 HAL Libraries start TX Dma transaction last
     PHAL_startTxfer(spi->tx_dma_cfg);
```

Comment:
Will PHAL_SPI_busy work if SPI1 and SPI2 are active at the same time? You may have to pass in which SPI you’re wanting to check.